File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/presentation/view/InventoryFragment.java
package br.com.alura.ceep.ui.coffemachine.presentation.view;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import br.com.alura.ceep.ui.coffemachine.R;
import br.com.alura.ceep.ui.coffemachine.domain.CoffeMachineDataItem;
import br.com.alura.ceep.ui.coffemachine.presentation.view.custom.CoffeAdapter;
import br.com.alura.ceep.ui.coffemachine.presentation.view.custom.ItemAdapter;

import com.google.android.material.textfield.TextInputLayout;

public class InventoryFragment extends Fragment {

  private TextInputLayout name;
  private TextView amount;
  private Button plus;
  private Button less;
  private Button save;
  private RecyclerView recyclerView;
  public static final String EXTRA_REPLY =
          "com.example.android.Coffe_Machine.REPLY";

  @Override
  public View onCreateView(LayoutInflater inflater, ViewGroup container,
      Bundle savedInstanceState) {
    return inflater.inflate(R.layout.inventory_fragment, container, false);
  }

  @Override public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    init(view);
    listeners();
    initList();
  }

  private void init(View view) {
    name = view.findViewById(R.id.name);
    amount = view.findViewById(R.id.amount);
    plus = view.findViewById(R.id.plus);
    less = view.findViewById(R.id.less);
    save = view.findViewById(R.id.save);
    recyclerView = view.findViewById(R.id.recyclerView);
  }

  private void listeners() {
    plus.setOnClickListener(v -> {
      String amountNotConverted = amount.getText().toString();
      int amountConverted = Integer.parseInt(amountNotConverted);
      int total = amountConverted + 1;
      String amountFormatted = String.valueOf(total);
      amount.setText(amountFormatted);
    });
    less.setOnClickListener(v -> {
      String amountNotConverted = amount.getText().toString();
      int amountConverted = Integer.parseInt(amountNotConverted);
      if (amountConverted > 0) {
        int total = amountConverted - 1;
        String amountFormatted = String.valueOf(total);
        amount.setText(amountFormatted);
      }
    });
    save.setOnClickListener(v -> {
      Toast.makeText(requireContext(), "Salvo com sucesso", Toast.LENGTH_SHORT).show();
        Intent Intent = new Intent();
          String capsule = amount.getText ().toString();
          Intent.putExtra (EXTRA_REPLY,capsule);
        })
    ;};

  private void initList() {
    recyclerView.setHasFixedSize(true);
    recyclerView.setLayoutManager(new LinearLayoutManager(requireContext()));
    CoffeMachineDataItem[] coffeMachineDataItem = new CoffeMachineDataItem[] {
        new CoffeMachineDataItem(
                "Ristretto",
                "",
                R.drawable.ristretto),
        new CoffeMachineDataItem (
                "Capuccino",
                "",
                R.drawable.capuccino),
    };
    ItemAdapter itemAdapter = new ItemAdapter (coffeMachineDataItem);
    recyclerView.setAdapter(itemAdapter);
    itemAdapter.notifyDataSetChanged();
  }
}

File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/presentation/view/InventoryFragment.kt
package br.com.alura.ceep.ui.coffemachine.presentation.view

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import br.com.alura.ceep.ui.coffemachine.R
import br.com.alura.ceep.ui.coffemachine.presentation.view.custom.ItemAdapter
import com.google.android.material.textfield.TextInputLayout

class InventoryFragment : Fragment() {
    private var name: TextInputLayout? = null
    private var amount: TextView? = null
    private var plus: Button? = null
    private var less: Button? = null
    private var save: Button? = null
    private lateinit var recyclerView: RecyclerView
//    private val viewModel: CoffesViewModel by activityViewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.inventory_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init(view)
        listeners()
        initList()
    }

    private fun init(view: View) {
        name = view.findViewById(R.id.name)
        amount = view.findViewById(R.id.amount)
        plus = view.findViewById(R.id.plus)
        less = view.findViewById(R.id.less)
        save = view.findViewById(R.id.save)
        recyclerView = view.findViewById(R.id.recyclerView)
    }

    private fun listeners() {
        plus!!.setOnClickListener { v: View? ->
            val amountNotConverted = amount!!.text.toString()
            val amountConverted = amountNotConverted.toInt()
            val total = amountConverted + 1
            val amountFormatted = total.toString()
            amount!!.text = amountFormatted
        }
        less!!.setOnClickListener { v: View? ->
            val amountNotConverted = amount!!.text.toString()
            val amountConverted = amountNotConverted.toInt()
            if (amountConverted > 0) {
                val total = amountConverted - 1
                val amountFormatted = total.toString()
                amount!!.text = amountFormatted
            }
        }
        save!!.setOnClickListener { v: View? ->
            Toast.makeText(requireContext(), "Salvo com sucesso", Toast.LENGTH_SHORT).show()
            val Intent = Intent()
            val capsule = amount!!.text.toString()
            Intent.putExtra(
                EXTRA_REPLY,
                capsule
            )
        }
    }

    private fun initList() {
        val itemAdapter = ItemAdapter()
        viewModel.list.observe(viewLifecycleOwner) { coffes ->
//            model.getAllCoffes()
            itemAdapter.list.addAll(coffes)
            itemAdapter.notifyDataSetChanged()
        }
//        recyclerView.setHasFixedSize(true)
        recyclerView.layoutManager = LinearLayoutManager(requireContext())
        recyclerView.adapter = itemAdapter
        itemAdapter.notifyDataSetChanged()

//        lifecycleScope.launch {
//            viewModel.getAllUsers()
//        }
//        val coffeMachineDataItem = arrayOf(
//            CoffeMachineDataItem(
//                "Ristretto",
//                "",
//                R.drawable.ristretto
//            ),
//            CoffeMachineDataItem(
//                "Capuccino",
//                "",
//                R.drawable.capuccino
//            )


    }
    companion object {
        const val EXTRA_REPLY = "com.example.android.Coffe_Machine.REPLY"
    }
}